examples/advanced_operations/add_smart_display_ad.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This adds a smart display campaign, an ad group, and a responsive display ad.
+
+More information about Smart Display campaigns can be found at:
+https://support.google.com/google-ads/answer/7020281
+"""
+
+
+import argparse
+import datetime
+import sys
+from uuid import uuid4
+
+import google.ads.google_ads.client
+import requests
+
+
+_date_format = '%Y%m%d'
+_marketing_image_url = 'https://goo.gl/3b9Wfh'
+_marketing_image_width = 600
+_marketing_image_height = 315
+_square_marketing_image_url = 'https://goo.gl/mtt54n'
+_square_marketing_image_size = 512
+
+
+def main(client, customer_id):
+    budget_resource_name = _create_budget(client, customer_id)
```

Comment:
I think it is cleaner to put all of main functionality into a single try block instead of try blocks for each service. I did that with my most recent sample and think we should adopt it going forward as it avoids code repetition.

Reply:
I disagree. While I'm not usually strict about excluding simple things like printing output, I'm in favor of the [style guideline recommendation](http://go/pystyle#exceptions-decision) of keeping try/except blocks small to make the cause of a given exception very clear.

Reply:
Well, we'll have to agree to disagree. Since these are all related API calls I think a single try block makes sense.